subsys/net/lib/nrf_cloud/src/nrf_cloud_pgps.c

```diff
@@ -1060,28 +1129,63 @@ static int flash_callback(uint8_t *buf, size_t len, size_t offset)
 void *flash_callback;
 #endif
 
-static int open_storage(uint32_t offset, bool preserve)
+static int open_flash(void)
 {
 	int err;
-	const struct device *flash_dev = DEVICE_DT_GET(DT_NODELABEL(flash_controller));
-	uint32_t block_offset = 0;
 
-	LOG_DBG("storage name: %s", flash_dev->name);
-	if (!device_is_ready(flash_dev)) {
-		LOG_ERR("Flash device not ready:'%s'", flash_dev->name);
-		return -ENODEV;
+#if defined(CONFIG_NRF_CLOUD_PGPS_STORAGE_PARTITION)
+	prediction_flash_dev = FLASH_AREA_DEVICE(PGPS);
+	flash_area_id = FLASH_AREA_ID(PGPS);
+#elif defined(CONFIG_NRF_CLOUD_PGPS_STORAGE_MCUBOOT_SECONDARY)
+	prediction_flash_dev = FLASH_AREA_DEVICE(MCUBOOT_SECONDARY);
+	flash_area_id = FLASH_AREA_ID(MCUBOOT_SECONDARY);
+#else
+	prediction_flash_dev = DEVICE_DT_GET(DT_NODELABEL(flash_controller));
+	flash_area_id = FLASH_AREA_ID(DT_NODELABEL(flash_controller));
+#endif
+
+	err = flash_area_open(flash_area_id, &prediction_flash_area);
+	if (err) {
+		prediction_flash_area = NULL;
+		LOG_ERR("Cannot access predictions using flash_area: %d", err);
```

Comment:
return here and remove `else`?
or is it useful to call `device_is_ready` below?  
will that ever succeed if `flash_area_open` fails?

Reply:
At some point during development of this feature, I noticed the main flash in the 9160 was failing to open with `flash_area_open`, so this is why I wrote it like this.  In that case, it didn't matter since I would access the flash with direct reads and `stream_flash_buffered_write`.

Now, however, it looks like `flash_area_open` works regardless, and it also does the `device_is_ready` check.  So, I can simplify this.